docs/deploying/ssl.html

```diff
@@ -0,0 +1,114 @@
+      <p class="block-text">Transport Layer Security (TLS) is a protocol for encrypting data that is sent over the internet. It provides three important features:</p>
+      <ul>
+        <li>Data Privacy: The data that is exchanged between a server and client is not visible to anyone else.</li>
+        <li>Data Integrity: The data that is exchanged between a server and client cannot be modified by anyone else.</li>
+        <li>Server Authenticity: The server can prove its identity to the client and so prove the origin of sent data.</li>
+    </ul>
+    <p class="block-text">TLS evolved out of Netscape’s Secure Sockets Layer (SSL) protocol in order to fix most of its security flaws.
+         The industry still uses the terms somewhat interchangeably for historical reasons.
+          Any web site that you visit starting with https:// rather than http:// is using TLS/SSL for communication between your browser and their server.</p>
+      <div class="underline"></div>
+      <h2 class="heading-2" id="prereq">Prerequisites</h2>
+      <ul class="plain-list">
+          <li><a onclick="setActiveSidebarElementForParent('create-server')" href="../getting-started/create-server.html#">Kitura Server:</a> Learn how to create one in our Getting Started guide.</li>
+          <li><a onclick="setActiveSidebarElementForParent('style-guide')" href="../getting-started/style-guide.html#">Kitura Style Guide:</a> Follow our guide to see how a production ready server should be structured.</li>
+          <li><a onclick="localStorage.setItem('package', 'FileKit'); setActiveSidebarElementForParent('filekit')" href="../getting-started/update-package.html#">Update Dependencies:</a> FileKit needs adding to the `Package.swift` file.</li>
+          <li>Install OpenSSL: In this guide, we will use OpenSSL to generate our self-signed certificate.</li>
+      </ul>
+      <p>OpenSSL is included by default on Linux. On MacOS we can install it using Homebrew:</p>
+      <pre><code>brew install openssl</code></pre>
+      <p>We then link it so we can use the OpenSSL CLI:</p>
+      <pre><code>ln -s /usr/local/opt/openssl/include/openssl /usr/local/include</code></pre>
+      <div class="underline"></div>
+      <h2 class="heading-2"><span class="blue-text">Step 1:</span> Create our certificate and key pair.</h2>
+      <p class="block-text">To enable TLS, a server needs a certificate and a corresponding secret key.
+           Certificates are files that bind together information about the identity of the owner of a site and the public half of an asymmetric key pair (usually RSA).
+           Certificates are usually digitally signed by a certificate authority (CA) who verifies that the identity information in the certificate is correct.
+           This creates a chain of certificates between the site owner certificate and a CA certificate and transitive trust.
+            Assuming that we trust the CA, we can trust the validity of the server certificate.</p>
```

Comment:
Are we going to tell the user anything about the fact that in some cases your cloud provider has in-built methods for implementing SSL as we don't want them to follow all this and find it wasn't needed?

Reply:
Andy, thanks for adding the new hint which clarifies this 👍